src/check/constrain/constraint/mod.rs

```diff
@@ -58,6 +60,31 @@ impl Constraint {
         Self::access(msg, expected, &Name::from(clss::BOOL), &StringName::from(function::TRUTHY))
     }
 
+    /// Create two constraints for collection:
+    ///
+    /// - A constraint which constraint a call to [ITER] with an iterator, which has an unknown type at this point.
+    ///   To over-bridge this, a temporary type name must be given using [helper_ty].
+    /// - A constraint which constraints [NEXT] call to temporary type with given [col_ty]
+    pub fn collection(msg: &str, expected: &Expected, col_ty: &Name, iter_ty: &Name) -> (Constraint, Constraint) {
```

Comment:
Perhaps rename to `iterable` since that is more representative of the constraints we are generating.
We are checking that this returns something akin to an iterator (we do duck-typing of sorts because we don't check that is actually an iterator we just check that it implements `__next__()`).